File: solutions/fsharp/rna-transcription/2/RnaTranscription.fs
module RnaTranscription

let toRna (dna: string) : string =
    let fromDna c =
        match c with
        | 'G' -> "C"
        | 'C' -> "G"
        | 'T' -> "A"
        | 'A' -> "U"
        | _ -> failwith "Invalid character"

    String.collect fromDna dna

File: solutions/ocaml/rna-transcription/2/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `U | `G | `C | `A ]

let to_rna = 
    let dna_to_rna = function 
    | `A -> `U 
    | `C -> `G 
    | `G -> `C 
    | `T -> `A
    in List.map dna_to_rna
